benchmark/benchmark.py

```diff
@@ -237,6 +238,72 @@ def _get_ec2_instance_id() -> Optional[str]:
 
     return instance_id
 
+class ResourceMonitoring():
+    def __init__(self, with_bwm: bool):
+        """Resource monitoring setup.
+
+        with_bwm: Whether to start bandwidth monitor tool `bwm-ng`.  Optional because it's not available
+        in the default AL2023 distro so you have to install it first."""
+        self.mpstat_process = None
+        self.bwm_ng_process = None
+        self.with_bwm = with_bwm
+        self.output_files = []
+
+    def _start(self) -> None:
+        log.debug("Starting resource monitors...")
+        self.mpstat_process = self._start_mpstat()
+        if self.with_bwm:
+            self.bwm_ng_process = self._start_bwm_ng()
+
+    def _close(self) -> None:
+        log.debug("Shutting down resource monitors...")
+        if self.mpstat_process:
+            self.mpstat_process.send_signal(signal.SIGINT)
+            self.mpstat_process.wait()
+        if self.bwm_ng_process:
+            self.bwm_ng_process.send_signal(signal.SIGINT)
+            self.bwm_ng_process.wait()
+        for output_file in self.output_files:
```

Comment:
This should be in a `finally` block to make sure the output files are always closed